### ISSUE TITLE
replace alerts with html modal dialogs

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-browser#1516]
One option: https://github.com/sweetalert2/sweetalert2
```
36K	sweetalert2.css (unminified)
80K	sweetalert2.js (unminified)
```
I would prefer something more lightweight, but this looks ok-light-enough.

Crucial:
 - I need to be able to `await` user response (including dialogs that only have "ok" option). A lot of code relies on `alert()` blocking other user interactions
 - option to fire either `ok` dialog or `ok / cancel` dialog.

Useful but not crucial:
 - custom html content, specifically links. I really like the html footer with a link here:
![image](https://user-images.githubusercontent.com/6306961/52163537-46db9600-26db-11e9-82a8-4a669265be29.png)

**Comment by tomholub** [comment]
todo:
 - [ ] donate to whichever solution we ended up using

**Comment by tomholub** [comment]
We don't due this for aesthetics - mostly for our automated tests. Currently, browser alert trips up puppeteer in unpredictable ways if it was not expected by the puppeteer code. One of many examples: https://github.com/GoogleChrome/puppeteer/issues/2481

Second reason is that users may accidentally block alerts, which breaks functionality.

Third is that a lot of functionality happens inside iframes, so it would be good to have the modal positioned over the iframe that causes it, and not block interaction with the host page (eg mail.google.com), in case user wants to navigate to another conversation, etc.

**Comment by tomholub** [comment]
## First stage

An easy, low risk page to test this out is the feedback page, so I suggest you start there. See `FlowCrypt Settings -> Need help?` -> after submitting the email, it will alert `Message sent! You will find your response in ...., check your email later. Thanks!`. You can replace this with a modal. Also, currently it will successfully send an empty message. This is not desired, so if the message is empty, you could show an error modal, asking the user to enter a message.

If the modal was a success notification, please `await` user confirmation (eg `ok` or whatever appropriate) and close the page.
If modal was error (missing msg), `await` user confirmation, then focus on appropriate form input.

Once that is done, I would like to review it to make sure these modals behave well, and that we are using the right library, etc.

## Second stage

Replace all error alerts in `js/composer.js` with this modal. This is more complex, and you may find the following of value: https://github.com/FlowCrypt/flowcrypt-browser/issues/1070#issuecomment-459972431

## Third stage

Bonus points if you can also appropriately adjust automated tests, though that is no small undertaking, and requires more guidance and env setup, so it's possibly out of scope for this task.

**Comment by jkdamilola** [comment]
@tomholub Is anyone working on this already? I think it will be a good start for me and I'm interested in working on it.

**Comment by tomholub** [comment]
@jkdamilola yup someone is evaluating / tackling this one. Good you asked. Please first write me at **tom@flowcrypt.com** - I would first like to review your previous code, talk about compensation etc - then we can find a good test issue for you.
